zetasql/fuzzing/zetasql_expression_proto_to_string.cc

```diff
@@ -0,0 +1,132 @@
+//
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include <string>
+
+#include "zetasql/fuzzing/common.h"
+#include "zetasql/fuzzing/zetasql_expression_grammar.pb.h"
+
+using namespace zetasql_expression_grammar;
+
+namespace zetasql_fuzzer {
+
+std::string HandleDefault() {
+  return "0";
+}
+
+// Forward declaration
+CONV_FN(Expression, expr);
+
```

Comment:
std::string ExpressionToString(const Expression& expr);

Losing the macro is much better in my opinion. It makes all design choices explicit, and it's more likely to be understood by IDEs and other code tools (including people who use vi /s and grep). Otherwise I have to keep checking what the macro does and making sure no one changed it to add side effects.

Reply:
We can pull back the CONV_FN macro back into the cc file. In this way people don't need to go to "common.h" to understand what it does. In the "to_string.h" we can write out the complete function name "ExpressionToString" as the public API is more likely to be studied. Again, CONV_FN seems to be just as a convenience shorthand since we expect to use it a lot.

Reply:
OK, it's a matter of style and your code. :)